compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -165,6 +165,25 @@ pub struct ImplHeader<'tcx> {
     pub predicates: Vec<Predicate<'tcx>>,
 }
 
+#[derive(
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    TyEncodable,
+    TyDecodable,
+    HashStable,
+    Debug,
+    TypeFoldable
+)]
+pub enum ImplicitBound {
```

Comment:
Why isn't this just a "Yes" or "No"? `Sized` should be inferred from the trait def id?

Reply:
This could potentially be extended to support auto-traits in general, but it can easily be changed.

Reply:
Changed to `Yes`.